common/src/test/java/gov/cms/ab2d/common/service/PerformanceTestingCoverageOperations.java

```diff
@@ -0,0 +1,313 @@
+        timing.sort((a, b) -> Long.valueOf(b - a).intValue());
+
+        long sumMillis = timing.stream().mapToLong(i -> i).sum();
+
+        System.out.println("Average millis per select: " + (sumMillis / (queries * threads)));
+        System.out.println("Max times " + timing.stream().limit(30).map(Objects::toString).collect(joining(", ")));
+    }
+
+    @Disabled
+    @DisplayName("delete previous search performance")
+    @Test
+    void deletePreviousSearch() {
+
+        coverageService.submitCoverageSearch(period.getId(), "testing");
+        CoverageSearchEvent inProgress1 = coverageService.startCoverageSearch(period.getId(), "testing");
+
+        coverageSearchEventRepo.saveAndFlush(inProgress1);
+        insertData(inProgress1, 100_000, 1, false);
+
+        coverageService.completeCoverageSearch(period.getId(), "testing");
+
+        coverageService.submitCoverageSearch(period.getId(), "testing");
+        CoverageSearchEvent inProgress2 = coverageService.startCoverageSearch(period.getId(), "testing");
+
+        coverageSearchEventRepo.saveAndFlush(inProgress2);
+        insertData(inProgress2, 100_000, 1, false);
+
+        coverageService.completeCoverageSearch(period.getId(), "testing");
+
+
+        System.out.println("Records present before delete " + coverageRepo.countByCoverageSearchEvent(inProgress1));
+
+        Instant start = Instant.now();
+        coverageService.deletePreviousSearch(period.getId());
+        Instant end = Instant.now();
+
+        System.out.println("Records present after delete " + coverageRepo.countByCoverageSearchEvent(inProgress1));
+
+        System.out.println("Time to delete previous search in milliseconds " + Duration.between(start, end).toMillis());
+    }
+
+
+    /**
+     * Load fake data into the database for stress testing reads
+     */
+    private void loadDBWithFakeData(int dataPoints) {
+
+        if (coverageRepo.count() == 0) {
+            CoverageSearchEvent inProgress = new CoverageSearchEvent();
+            inProgress.setOldStatus(JobStatus.SUBMITTED);
+            inProgress.setNewStatus(JobStatus.IN_PROGRESS);
+            inProgress.setDescription("testing");
+            inProgress.setCoveragePeriod(period);
+
+            coverageSearchEventRepo.saveAndFlush(inProgress);
+
+            List<Long> timings = insertData(inProgress, dataPoints, 1, false);
+            System.out.println("Insertion took " + timings.get(0));
+        }
+    }
+
+    public static class BeneficiaryIdSupplier implements Supplier<String> {
+
+        private int id = 0;
+
+        public BeneficiaryIdSupplier() {}
+
+        public String get() {
+            return "test-" + id++;
+        }
+    }
+
+    private List<Long> insertData(CoverageSearchEvent inProgress, int dataPoints, int experiments, boolean erase) {
+        BeneficiaryIdSupplier supplier = new BeneficiaryIdSupplier();
+
+        List<Long> timings = new ArrayList<>();
+
+        int i = 0;
+        while (i < experiments) {
```

Comment:
Another for loop candidate.  Please use *idx* or some other variable instead of 1 letter ones.

Reply:
Converted